PerimeterXModule/Internals/VerificationHandlers/DefaultVerificationHandler.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net;
+using System.Web;
+
+namespace PerimeterX
+{
+    public class DefaultVerificationHandler : VerificationHandler
+    {
+        private IActivityReporter activityReporter;
+
+        public DefaultVerificationHandler(IActivityReporter activityReporter)
+        {
+            this.activityReporter = activityReporter;
+        }
+
+        public bool HandleVerificatoin(PXConfigurationWrapper pxConfig, PxContext pxCtx, HttpApplication application)
+        {
+            int score = pxCtx.Score;
+
+            if (score < pxConfig.BlockingScore || pxConfig.MonitorMode){
+                Debug.WriteLine("Request was verified, passing request, score {0} mintor mode {1}", score, pxConfig.MonitorMode);
+                PostPageRequestedActivity(pxCtx, pxConfig);
+                return true;
+            }
+
+			Debug.WriteLine("Request was not verified, blocking request, score {0}", score, pxConfig.MonitorMode);
+            PostBlockActivity(pxCtx, pxConfig);
+
+            application.Response.StatusCode = (int)HttpStatusCode.Forbidden;
+            application.Response.TrySkipIisCustomErrors = true;
+			//End request
+			if (pxConfig.SuppressContentBlock)
+			{
+				pxCtx.ApplicationContext.Response.SuppressContent = true;
+			}
+
+            else
+            {
+                var html = ResponseBlockPage(pxCtx, pxConfig);
+                application.Response.Write(html);
+            }
+
+            application.CompleteRequest();
+            return false;
+		}
+
+		private void PostActivity(PxContext pxCtx, PXConfigurationWrapper pxConfig, string eventType, ActivityDetails details = null)
+		{
+			var activity = new Activity
+			{
+				Type = eventType,
+				Timestamp = Math.Round(DateTime.UtcNow.ToUniversalTime().Subtract(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)).TotalMilliseconds, MidpointRounding.AwayFromZero),
```

Comment:
is there no better way of doing the timestamp calculation then using the 1970 hard coded date?